static/js/compose.js

```diff
@@ -402,6 +405,37 @@ function compose_error(error_text, bad_input) {
     }
 }
 
+function send_typing_notification_ajax(recipients, operation) {
+    channel.post({
+        url: '/api/v1/typing',
+        data: {
+          to: recipients,
+          op: operation,
+        },
+        success: console.log,
+        error: function (xhr, error_type) {
+            var response = channel.xhr_error_message("Error sending typing notification", xhr);
+            console.log(response);
+        }
+    });
+}
+
+function send_typing_started_notification() {
+    function check_and_send() {
+        if (compose.message_content() && compose.recipient()) {
+            console.log("Going to send typing started notification to " + compose.recipient());
+            send_typing_notification_ajax(compose.recipient(), 'start');
+        }
+    }
+    check_and_send();
+    setInterval(check_and_send, 15000);
```

Comment:
Let's put all the values like `15000` in constants at the top of the file.  Might actually make sense to create a new `typing.js` module for clarity; `compose.js` is pretty big and I think it'd be nice to have the frontend code for this all together.